src/Analysis/Ast/Impl/Types/Definitions/IPythonClassType.cs

```diff
@@ -41,6 +41,6 @@ public interface IPythonClassType : IPythonClassMember, IGenericType {
         /// If class is created off generic template, represents
         /// pairs of the generic parameter / actual supplied type.
         /// </summary>
-        IReadOnlyDictionary<IGenericTypeParameter, IPythonType> GenericParameters { get; }
+        IReadOnlyDictionary<IGenericTypeParameter, IPythonType> ActualGenericParameters { get; }
```

Comment:
This name is really ambiguous. Does it mean the pairs of parameters where a specific type was supplied? For example T -> int?

We can also have pairs like T -> K where they are both generic parameters, how does the field look like in that case? 

When I was refactoring this, I was thinking of switching out `Parameters` to `GenericParametersToFill` and leaving `GenericParameters` the way it was.

Reply:
It reflects what the comments says "represents pairs of the generic parameter / actual supplied type". Sort of like in a function there are 'formal' and 'actual' arguments. So we could make `FormalGenericParameters`

Reply:
I should have updated those comments then - `GenericParameters` holds all pairs of generic parameter to types, regardless if they were supplied or not. 

If the types haven't been supplied then the value is simply the generic type parameter.

So the dictionary may look like:
T -> T
K -> int

etc.

T has not been given a value but K has.

But `Parameters` will be a list of the generic parameters that still need values, i.e the values of the dictionary that are of type `IGenericTypeParameter`.